### PR TITLE
Webpack setup

### PR DESCRIPTION
**Pull request by patriciachrysy**
# 👋 Micronaut Reviewer 

[To Do list: list structure](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m1_list_structure.md)

Today, i'm working on the project first step : 

### Project requirements
- [x] Set up a new project with webpack that is based on the webpack exercise you have already completed.
- [x] Create an **index.html** file and write your HTML markup here. Create an empty **To Do List** placeholder (`<div>` or `<ul>` element). The **index.html** file must be set as a template using the *HTML Webpack Plugin*.
- [x] Create an **index.js** file and set an array of some simple *to do tasks* (array of objects). Each task object should contain three keys: 
  1. description [`string`].
  2. completed [`bool`].
  3. index: [`number`].
- [x] Write a function to iterate over the tasks array and populate an HTML list item element for each task.
- [x] On page load render the dynamically created list of tasks in the dedicated placeholder. The list should apear in order of the `index` values for each task.
- [x] Create a **style.css** and set rules for the **To Do List**. CSS must be loaded by *Webpack Style/CSS Loader*. Your list should be a clone of the part of the minimalist project captured in the video below.

[Capstone intro](https://www.youtube.com/watch?v=AcUd-_Yjjqg)
- [x] All your source files (**index.html**, **index.js** and **style.css**) must be located in */src* directory and your distribution files will be generated by webpack and served by *webpack dev server* from */dist* folder.

Happy Review :)

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+  form.appendChild(submitButton);
+
+  formDiv.appendChild(form);
+
+  return formDiv;
+};
+
+const listTasks = () => {
+  const sortedTasks = toDoTasks.sort((a, b) => a.index > b.index);
```
Comment (D4nielJ) [inline src/index.js]: - [x] I think sort is a bit special, it does not compare objects as other functions, instead, it needs the returned value to be a number. You can see the details in this table: 

image

[You can read more about it here. ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort)

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  title.innerText = "Today's To Do";
+  titleDiv.appendChild(title);
+  containerDiv.appendChild(titleDiv);
+
+  containerDiv.appendChild(displayForm());
+
+  // eslint-disable-next-line no-restricted-syntax
+  for (const task of sortedTasks) {
+    containerDiv.appendChild(displayTask(task));
+  }
```
Comment (D4nielJ) [inline src/index.js]: - [x] Disabling eslint rules defeats the purpose of the linters in the first place. Instead, you can try rewriting this chunk of code with a accepted sintax.

hint: `sortedTasks.map((task) => {do something with the task})`

**Review by D4nielJ — CHANGES_REQUESTED** [review]
## Hello @patriciachrysy, 

You're doing great! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights :fire: 
- Your UI looks on point! :rocket: :sparkles: 
- You render dynamically each task on the array.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

## Happy coding and stay awesome! 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can always find me on slack!

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (patriciachrysy) [reply src/index.js]: Done

File: src/index.js (hunk 2 above)
Comment (patriciachrysy) [reply src/index.js]: Thanks for the tip

**Review by ArthurGC — APPROVED** [review]
# Approved project!  💯 ✅ 🥳 🎈 🎉
Hi  @patriciachrysy 

Amazing job implementing suggestions from the previous reviewer! 💯 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Aspects to improve ♻️
_Check the comments under the review._

- [OPTIONAL] It would be better to fix the broken links in your README file (Live Preview and MIT). If you don't have a live preview link, you should remove it from the README until you get one.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**